lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -420,15 +420,35 @@ class MongoClientInterface {
              * code is the one related to mentionned issue.
              */
             if (err) {
-                if (!err.index || err.index !== 1
-                 || !err.code || err.code !== 11000) {
-                    log.error(
-                        'putObjectVerCase1: error putting object version',
-                    { error: err.errmsg });
+                if (err.code === 11000) {
+                    log.debug('putObjectVerCase1: error putting object version', {
+                        code: err.code,
+                        error: err.errmsg,
+                        isRetry: isRetry ? 'true' : 'false',
+                    });
+                    let count = err.result.upsertedCount;
+                    if (typeof count !== 'number') {
+                        count = err.result.nUpserted;
+                    }
+                    if (typeof count === 'number' && count !== 1) {
+                        // This may be a race condition, when two different S3 Connector try to put the same version id
+                        if (!isRetry) {
+                            // retrying with a new version id
+                            return process.nextTick(() =>
+                                this.putObjectVerCase1(c, bucketName, objName, objVal, params, log, cb, true));
+                        }
+                        log.error('putObjectVerCase1: race condition upsetting versionId', {
```

Comment:
"upsetting" -> "setting" ?

Reply:
no it's upsetting value in mongodb ^^

Reply:
I'm not sure if "upsetting" means what it's supposed to mean here (kind of bumping the version), hence my comment, but since I'm not a native speaker I can definitely be wrong and would let this confirmed by a native speaker :us:

Reply:
or do you mean "upsert" maybe?

Reply:
true, that's actually wrong, it's upserting :)